engines/sci/graphics/plane32.h

```diff
@@ -0,0 +1,465 @@
+#include "sci/graphics/screen_item32.h"
+
+namespace Sci {
+enum PlaneType {
+	kPlaneTypeColored     = 0,
+	kPlaneTypePicture     = 1,
+	kPlaneTypeTransparent = 2,
+	kPlaneTypeOpaque      = 3
+};
+
+enum PlanePictureCodes {
+	// NOTE: Any value at or below 65532 means the plane
+	// is a kPlaneTypePicture.
+	kPlanePic            = 65532,
+	kPlanePicOpaque      = 65533,
+	kPlanePicTransparent = 65534,
+	kPlanePicColored     = 65535
+};
+
+#pragma mark -
+#pragma mark RectList
+
+typedef StablePointerArray<Common::Rect, 200> RectListBase;
+class RectList : public RectListBase {
+public:
+	void add(const Common::Rect &rect) {
+		RectListBase::add(new Common::Rect(rect));
+	}
+};
+
+#pragma mark -
+#pragma mark DrawList
+
+struct DrawItem {
+	ScreenItem *screenItem;
+	Common::Rect rect;
+
+	inline bool operator<(const DrawItem &other) const {
+		return *screenItem < *other.screenItem;
+	}
+};
+
+typedef StablePointerArray<DrawItem, 250> DrawListBase;
+class DrawList : public DrawListBase {
+private:
+	inline static bool sortHelper(const DrawItem *a, const DrawItem *b) {
+		return *a < *b;
+	}
+public:
+	void add(ScreenItem *screenItem, const Common::Rect &rect);
+	inline void sort() {
+		pack();
+		Common::sort(begin(), end(), sortHelper);
+	}
+};
+
+class PlaneList;
+
+#pragma mark -
+#pragma mark Plane
+
+/**
+ * A plane is a grouped layer of screen items.
+ */
+class Plane {
+private:
+	/**
+	 * A serial used for planes that are generated inside
+	 * the graphics engine, rather than the interpreter.
+	 */
+	static uint16 _nextObjectId;
+
+	/**
+	 * The dimensions of the plane, in game script
+	 * coordinates.
+	 * TODO: These are never used and are always
+	 * scriptWidth x scriptHeight in SCI engine? The actual
+	 * dimensions of the plane are always in
+	 * gameRect/planeRect.
+	 */
+	int16 _width, _height;
+
+	/**
+	 * For planes that are used to render picture data, the
+	 * resource ID of the picture to be displayed. This
+	 * value may also be one of the special
+	 * PlanePictureCodes, in which case the plane becomes a
+	 * non-picture plane.
+	 */
+	GuiResourceId _pictureId;
+
+	/**
+	 * Whether or not the contents of picture planes should
+	 * be drawn horizontally mirrored. Only applies to
+	 * planes of type kPlaneTypePicture.
+	 */
+	bool _mirrored;
+
+	/**
+	 * Whether the picture ID for this plane has changed.
+	 * This flag is set when the plane is created or updated
+	 * from a VM object, and is cleared when the plane is
+	 * synchronised to another plane (which calls
+	 * changePic).
+	 */
+	bool _pictureChanged; // ?
+
+	// TODO: Are these ever actually used?
+	int _field_34, _field_38; // probably a point or ratio
+	int _field_3C, _field_40; // probably a point or ratio
+
+	/**
+	 * Converts the dimensions of the game rect used by
+	 * scripts to the dimensions of the plane rect used to
+	 * render content to the screen. Coordinates with
+	 * remainders are rounded up to the next whole pixel.
+	 */
```

Comment:
No, kLocalToGlobal takes the position of a coordinate relative to a plane and changes it to a coordinate relative to the top-left corner of the screen always in game script coordinates. This converts the game script coordinate rect of a plane into its screen coordinate rect.